cmd/gaia/app/app.go

```diff
@@ -169,32 +176,150 @@ func NewGenesisAccount(acc *auth.BaseAccount) GenesisAccount {
 	}
 }
 
-// convert GenesisAccount to GaiaAccount
+// convert GenesisAccount to auth.BaseAccount
 func (ga *GenesisAccount) ToAccount() (acc *auth.BaseAccount) {
 	return &auth.BaseAccount{
 		Address: ga.Address,
 		Coins:   ga.Coins.Sort(),
 	}
 }
 
-// DefaultGenAppState expects two args: an account address
-// and a coin denomination, and gives lots of coins to that address.
-func DefaultGenAppState(args []string, addr sdk.Address, coinDenom string) (json.RawMessage, error) {
+var (
+	flagAccounts = "accounts"
+	flagChainID  = "chain-id"
+	flagOWK      = "overwrite-keys"
+)
+
+// get app init parameters for server init command
+func GaiaAppInit() server.AppInit {
+	fs := pflag.NewFlagSet("", pflag.ContinueOnError)
+	fs.String(flagAccounts, "foobar-10fermion,10baz-true", "genesis accounts in form: name1-coins-isval:name2-coins-isval:...")
```

Comment:
For CLI testing it's  *much* easier to have generation within the init command - more clear/concise for comprehension of the tests  ..
I disagree that we want to discourage automation - it's critical for the maintainability of the tests

Reply:
CLI testing is fine, I mean for external setups like a network deployment automation script for AWS or something. What seems silly would be someone writing another script to translate a structured map of genesis accounts to a CLI flag, which then gets parsed here and turned back into the structured `genesis.json`. Having this as an additional option doesn't hurt though, such a setup can still write the genesis file if they like.

Reply:
> What seems silly would be someone writing another script to translate a structured map of genesis accounts to a CLI flag, which then gets parsed here and turned back into the structured genesis.json

oh yeah makes sense - BUT did you know you don't even have to create any accounts with this flag? The accounts can be just added directly to the piece.json file after it's created - alternatively we could also have the account take a json file instead of the flag as it's been formatted - probably be easier in a deployment scenario

Reply:
This comment is now irrelevant after a refactor of this PR - we just are opinionated in this init and don't take any coin information